File: KoRadio/KoRadio.Services/Database/Tender.cs
using System;
using System.Collections.Generic;

namespace KoRadio.Services.Database;

public partial class Tender
{
    public int TenderId { get; set; }

    public DateTime JobDate { get; set; }

    public string JobDescription { get; set; } = null!;

    public byte[]? Image { get; set; }

    public bool IsFinalized { get; set; }

    public int? UserId { get; set; }

    public int? FreelancerId { get; set; }

    public int? CompanyId { get; set; }

    public bool IsFreelancer { get; set; }

    public virtual Company? Company { get; set; }

    public virtual Freelancer? Freelancer { get; set; }

    public virtual ICollection<TenderService> TenderServices { get; set; } = new List<TenderService>();

    public virtual User? User { get; set; }
}

File: KoRadio/UI/ko_radio_desktop/lib/models/tender.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:ko_radio_desktop/models/company.dart';
import 'package:ko_radio_desktop/models/freelancer.dart';
import 'package:ko_radio_desktop/models/tender_bids.dart';
import 'package:ko_radio_desktop/models/tender_service.dart';
import 'package:ko_radio_desktop/models/user.dart';
part 'tender.g.dart';

@JsonSerializable()
class Tender{
  int tenderId;
  DateTime? jobDate; 
  String? jobDescription;
  String? image;
  bool? isFinalized;
  int? userId;
  int? freelancerId;
  int? companyId;
  bool? isFreelancer;
  Company? company;
  User? user;
  Freelancer? freelancer;
  List<TenderBid>? tenderBids;
  List<TenderService>? tenderServices;

  Tender(this.tenderId);

  factory Tender.fromJson(Map<String, dynamic> json) => _$TenderFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$TenderToJson(this);

}